python/lsst/ctrl/bps/generic_workflow.py

```diff
@@ -0,0 +1,465 @@
+        self.dataset_ref = dataset_ref
+        self.dest_uri = dest_uri
+        self.logical_file_name = logical_file_name
+
+    __slots__ = ('name', 'wms_transfer', 'dataset_ref', 'src_uri', 'dest_uri', 'logical_file_name')
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __str__(self):
+        return f"GenericWorkflowJob(name={self.name})"
+
+
+@dataclasses.dataclass
+class GenericWorkflowJob:
+    """Information about a job that may be needed by various workflow
+    management services.
+    """
+    name: str
+    label: Optional[str]
+    cmdline: Optional[str]
+    request_memory: Optional[int]    # MB
+    request_cpus: Optional[int]       # cores
+    request_disk: Optional[int]      # MB
+    request_walltime: Optional[str]  # minutes
+    compute_site: Optional[str]
+    mail_to: Optional[str]
+    when_to_mail: Optional[str]
+    number_of_retries: Optional[int]
+    retry_unless_exit: Optional[int]
+    abort_on_value: Optional[int]
+    abort_return_value: Optional[int]
+    priority: Optional[str]
+    category: Optional[str]
+    pre_cmdline: Optional[str]
+    post_cmdline: Optional[str]
+    profile: Optional[dict]
+    attrs: Optional[dict]
+    environment: Optional[dict]
+    quantum_graph: Optional[QuantumGraph]
+    quanta_summary: Optional[str]
+
+    # As of python 3.7.8, can't use __slots__ if give default values, so writing own __init__
+    def __init__(self, name: str):
+        self.name = name
+        self.label = None
+        self.cmdline = None
+        self.request_memory = None
+        self.request_cpus = None
+        self.request_disk = None
+        self.request_walltime = None
+        self.compute_site = None
+        self.mail_to = None
+        self.when_to_mail = None
+        self.number_of_retries = None
+        self.retry_unless_exit = None
+        self.abort_on_value = None
+        self.abort_return_value = None
+        self.priority = None
+        self.category = None
+        self.pre_cmdline = None
+        self.post_cmdline = None
+        self.profile = {}
+        self.attrs = {}
+        self.environment = {}
+        self.quantum_graph = None
+        self.quanta_summary = ""
+
+    __slots__ = ('name', 'label', 'mail_to', 'when_to_mail', 'cmdline', 'request_memory', 'request_cpus',
+                 'request_disk', 'request_walltime', 'compute_site', 'environment', 'number_of_retries',
+                 'retry_unless_exit', 'abort_on_value', 'abort_return_value', 'priority',
+                 'category', 'pre_cmdline', 'post_cmdline', 'profile', 'attrs',
+                 'quantum_graph', 'quanta_summary')
+
+    def __hash__(self):
+        return hash(self.name)
+
+
+class GenericWorkflow(nx.DiGraph):
+    """A generic representation of a workflow used to submit to specific
+    workflow management systems.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of generic workflow.
+    incoming_graph_data : ` `, optional
+    attr : optional
```

Comment:
I guess the descriptions are optional? (I'll stop making noise about this)

Reply:
This documentation is not following the standard. It needs the description text and the type text.

Reply:
I'm pretty sure they aren't optional.  It just means I got distracted at the time and didn't go back and find them. 
 And neither flake8 nor pylint complained about them missing.  I will check them all.  Thanks.